larray_editor/arraymodel.py

```diff
@@ -349,13 +356,8 @@ def data(self, index, role=Qt.DisplayRole):
                 if self.bg_value is None:
                     return self.bg_gradient[float(self.color_func(value))]
                 else:
-                    bg_value = self.bg_value
-                    x, y = index.row(), index.column()
-                    # FIXME: this is buggy on filtered data. We should change
-                    # bg_value when changing the filter.
-                    idx = y + x * bg_value.shape[-1]
-                    value = bg_value.data.flat[idx]
-                    return self.bg_gradient[value]
+                    i, j = index.row(), index.column()
+                    return self.bg_gradient[self.bg_value[i][j]]
```

Comment:
why not use `self.bg_value[i, j]`?

Reply:
Forgot that bg_value is a Numpy ndarray.